src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriverAdapterUseControl.java

```diff
@@ -0,0 +1,59 @@
+package edu.kis.powp.jobs2d.drivers.adapter;
+
+import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.file.DataFile;
+
+import static java.lang.Math.pow;
+import static java.lang.Math.sqrt;
+
+import java.io.FileNotFoundException;
+
+public class LineDriverAdapterUseControl implements Job2dDriver {
+
+    private Job2dDriver job2dDriver;
+
+    private double distance;
+    private int prevX, prevY;
+
+    public LineDriverAdapterUseControl(Job2dDriver job2dDriver) throws FileNotFoundException {
+        this.job2dDriver = job2dDriver;
+        DataFile dataFile = new DataFile(this);
```

Comment:
DataFile option should be extracted outside driver.
Initial distance should be set using setter.